pubsub/gcloud/aio/pubsub/subscriber.py

```diff
@@ -215,22 +224,27 @@ async def _execute_callback(message: SubscriberMessage,
                                 metrics_client: MetricsAgent
                                 ) -> None:
         try:
-            start = time.perf_counter()
-            await callback(message)
-            await ack_queue.put(message.ack_id)
-            metrics_client.increment('pubsub.consumer.succeeded')
-            metrics_client.histogram('pubsub.consumer.latency.runtime',
-                                     time.perf_counter() - start)
+            with PROMETHEUS.consume_latency.labels(aspect='runtime').time():
+                start = time.perf_counter()
+                await callback(message)
+                await ack_queue.put(message.ack_id)
+                metrics_client.increment('pubsub.consumer.succeeded')
+                metrics_client.histogram('pubsub.consumer.latency.runtime',
+                                         time.perf_counter() - start)
+                PROMETHEUS.consume.labels(outcome='succeeded').inc()
```

Comment:
nit: now that we won't need to carry around the `start` values, I wonder if these sorts of metrics would fit better in an eg. `else:` block. We probably don't want the time it takes to write the "success" metric to be included in the "consume latency", for example.

Reply:
Agreed, same goes for the receive latency (which was re-calculated for Prometheus). What about something like this?

```python
        try:
            start = time.perf_counter()
            with metrics.CONSUME_LATENCY.labels(aspect='runtime').time():
                await callback(message)
                await ack_queue.put(message.ack_id)
            metrics_client.histogram('pubsub.consumer.latency.runtime',
                                     time.perf_counter() - start)
            metrics_client.increment('pubsub.consumer.succeeded')
            metrics.CONSUME.labels(outcome='succeeded').inc()
        ...
```

Reply:
Yeah, that seems reasonable to me!

re: deprecation. Yeah, in fact we should probably disable them sooner rather than later to give more folks time to migrate. Thoughts on adding a `warrnings.warn` for when clients pass in a `metrics_client` as part of this PR?

Reply:
A deprecation warning sounds like a good idea to me!